### PR TITLE
Flatten left table of contents for Runtime and Provider

### PR DESCRIPTION
**Pull request by Eric-Arellano**
Closes https://github.com/Qiskit/documentation/issues/1212. As explained there, this is because:

1. Qiskit SDK is getting a bigger revamp in https://github.com/Qiskit/documentation/issues/1211 and it will also flatten submodules, so we want to be consistent.
2. The module list is so short that it's useful to have the navbar be flatter. It makes things less crowded and more discoverable.

Note that some of the Runtime provider pages aren't very useful...I opened https://github.com/Qiskit/documentation/issues/1240 to track that.

For now, this is implemented as the `Pkg` constructor setting a boolean argument for whether modules should be nested or not. I'm coming up with an alternative—but similar factoring—while working on https://github.com/Qiskit/documentation/issues/1211. This PR is ~prework.

Screenshot 2024-04-24 at 4 02 34 PM

Screenshot 2024-04-24 at 4 02 10 PM

### CODE REVIEW

**Comment by Eric-Arellano** [comment]
@arnaucasau @frankharkins this got sign off from Abby M, Kevin T, and Jessie, so this review is ensuring the implementation is good.

File: scripts/lib/api/generateToc.ts [hunk 1]
```diff
@@ -111,12 +112,18 @@ function addItemsToModules(
 }
 
 function getNestedTocModulesSorted(
+  pkg: Pkg,
   tocModules: TocEntry[],
   tocModulesByTitle: Dictionary<TocEntry>,
   tocModuleTitles: string[],
 ): TocEntry[] {
-  const nestedTocModules: TocEntry[] = [];
+  // Most packages don't nest submodules because there module list is so small,
```
Comment (Eric-Arellano) [inline scripts/lib/api/generateToc.ts:120]:
```suggestion
  // Most packages don't nest submodules because their module list is so small,
```

**Review by frankharkins — APPROVED** [review]
Thanks! Code makes sense and preview looks good.

Something we might want to think about (out of scope of this PR) is how we handle these long names on the sidebar. One option is to remove the `qiskit_ibm_runtime.` from each module name, but design might have a better solution.

Screenshot 2024-04-25 at 12 10 55

**Comment by Eric-Arellano** [comment]
> One option is to remove the qiskit_ibm_runtime. from each module name, but design might have a better solution.

I don't think we could do that though because the top level module `qiskit_ibm_runtime` has a lot of imports and we couldn't use the empty string 🤔 I think it'd be confusing if we had `qiskit_ibm_runtime` then `fake_provider` - people might think you could do `import fake_provider`.

**Comment by frankharkins** [comment]
Ok, let's see if design have a better idea

**Pull request merged**